js/ui/components/clubhouse.js

```diff
@@ -0,0 +1,323 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+//
+// Copyright (C) 2018 Endless Mobile, Inc.
+//
+// Licensed under the GNU General Public License Version 2
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// Author: Joaquim Rocha <jrocha@endlessm.com>
+//
+
+const Clutter = imports.gi.Clutter;
+const Flatpak = imports.gi.Flatpak
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const Shell = imports.gi.Shell;
+const Signals = imports.signals;
+const St = imports.gi.St;
+
+const GnomeSession = imports.misc.gnomeSession;
+const Lang = imports.lang;
+const Main = imports.ui.main;
+const MessageList = imports.ui.messageList;
+const MessageTray = imports.ui.messageTray;
+const Monitor = imports.ui.monitor;
+const NotificationDaemon = imports.ui.notificationDaemon;
+
+const sessionBus = Gio.bus_get_sync(Gio.BusType.SESSION, null);
+
+const CLUBHOUSE_ID = 'com.endlessm.Clubhouse';
+const CLUBHOUSE_DBUS_OBJ_PATH = '/com/endlessm/Clubhouse';
+
+var ClubhouseNotificationBanner = new Lang.Class({
+    Name: 'ClubhouseNotificationBanner',
+    Extends: MessageTray.NotificationBanner,
+
+    _init: function(notification) {
+        this.parent(notification);
+
+        // We don't have an "unexpanded" state for now
+        this.expanded = true;
+        this._actionBin.visible = true;
+
+        this.actor.can_focus = true;
+        this.actor.track_hover = true;
+
+        // Override the style name because this is a not a regular notification
+        this.actor.remove_style_class_name('notification-banner');
+        this.actor.add_style_class_name('clubhouse-quest-view');
+        this._iconBin.add_style_class_name('clubhouse-quest-view-icon-bin');
+
+        // Always wrap the body's text
+        this.bodyLabel.actor.clutter_text.line_wrap = true;
+        this.bodyLabel.actor.add_style_class_name('clubhouse-quest-view-label');
+
+        this._closeButton.connect('clicked', Lang.bind(this, function() {
+            this.emit('dismiss');
```

Comment:
I did this because I noticed that the base class destroys the notification when the closed signal is emitted, and in the beginning I was creating a new banner every time but wanted to keep only one notification. I have tried now and since I keep only one notification and one banner, I can reuse the `close` signal. Thanks!